examples/traffic-light.manifold

```diff
@@ -0,0 +1,41 @@
+* = import Manifold.core
+* = import Manifold.digital
+
+Type TrafficLight = (
+  green:Boolean  = false,
+  yellow:Boolean = false,
+  red:Boolean    = false
+)
+
+GREEN_SECONDS  = 10
+YELLOW_SECONDS = 3
+TOTAL_SECONDS  = 2 * (GREEN_SECONDS + YELLOW_SECONDS)
+
+seconds = Manifold.Digital.count(hz: 1, mod: TOTAL_SECONDS)
+
+Manifold.Digital.if(
+  (seconds < GREEN_SECONDS, -> {
+    north_light = south_light = (green: true)
```

Comment:
It's probably unambiguous to have `-> {` in certain places, like after `,` and `(`.
In some cases, would it be ambiguous for a parser, like `a -> b` could be calling a on `-> b` or a function from `a` to `b`.

Also, wouldn't these be local to the function scope, if `-> {` is a regular Manifold function? They're used but not declared in the global scope.